modules/nwtc-library/src/NWTC_IO.f90

```diff
@@ -1475,94 +1475,105 @@ SUBROUTINE AllR16Ary5 (  Ary, AryDim1, AryDim2, AryDim3, AryDim4, AryDim5, Descr
-
-   CHARACTER(*), INTENT(INOUT)          :: InputFile                                    !< The name of the input file specified on the command line. (first non-flag argument)
-   CHARACTER(*), INTENT(  OUT),OPTIONAL :: Arg2                                         !< an optional 2nd non-flag argument from command line
-   CHARACTER(*), INTENT(  OUT),OPTIONAL :: Flag                                         !< an optional flag argument (e.g., restart); the last argument starting with a switch character. 
+   CHARACTER(*), INTENT(INOUT)           :: InputFile          !< The name of the input file specified on the command line.
+   INTEGER,      INTENT(  OUT), OPTIONAL :: ErrStat            !< An optional argument for catching errors; if present, program does not abort on error.
+   CHARACTER(*), INTENT(  OUT), OPTIONAL :: Arg2               !< An optional 2nd non-flag argument from command line.
+   CHARACTER(*), INTENT(  OUT), OPTIONAL :: Flag               !< An optional flag argument; the first argument starting with a switch character. 
 
       ! Local declarations:
-
-   INTEGER                              :: IArg                                         ! The argument number.
-   INTEGER                              :: NumArg                                       ! The number of arguments on the command line.
-                                        
-   INTEGER                              :: Error                                        ! Error Status: indicates if there was an error getting an argument.
-   LOGICAL                              :: FirstArg                                     ! flag to determine if it's the first non-switch argument
-   CHARACTER(LEN(InputFile))            :: Arg                                          ! A command-line argument.
-   
+   INTEGER                               :: NumArg             ! The number of arguments on the command line.
+   CHARACTER(LEN(InputFile))             :: Arg                ! A command-line argument.
 
       ! Find out how many arguments were entered on the command line.
-   NumArg   = COMMAND_ARGUMENT_COUNT()
-   FirstArg = .TRUE.
+   NumArg = COMMAND_ARGUMENT_COUNT()
 
-   IF ( PRESENT(Arg2) ) Arg2 = ""
-   IF ( PRESENT(flag) ) flag = ""
+      ! Early return if no arguments given
+   IF ( NumArg == 0 ) CALL INVALID_SYNTAX( 'no command-line arguments given.' )
 
-      ! Parse them.
+      ! Parse the first argument; this is either a flag or the input file
+   CALL GET_COMMAND_LINE_ARG( 1, Arg, PRESENT(ErrStat))
 
-   IF ( NumArg .GT. 0 ) THEN
+      ! If the first argument is not one of the switch characters, then assume its the input file
+   IF ( Arg(1:1) /= SwChar .AND. Arg(1:1) /= '-' ) THEN
+      InputFile = Arg
 
-      DO IArg=1,NumArg
+      ! Parse the additional second argument, if it exists
+      IF ( NumArg == 2 .AND. PRESENT(Arg2) ) CALL GET_COMMAND_LINE_ARG( 2, Arg2, PRESENT(ErrStat))
```

Comment:
Your second argument here (`Arg2`) may or may not be a flag.

I would say that you need to allow for at least 
```
executable.exe -Flag FileName Arg2
executable.exe -Flag FileName
executable.exe FileName Arg2
executable.exe FileName
executable.exe
```
which I think you mostly have here. The previous logic also allowed the flags to be in any position on the line, like 
```
executable.exe FileName -Flag
executable.exe FileName -Flag Arg2
executable.exe FileName Arg2 -IgnoredFlagUnlessItIsHELP -Flag
```
While it is probably not necessary to reproduce that exact logic, it should be documented that it has changed, particularly since you won't be able to throw the `-h` flag anywhere except in the first argument.